torchtune/models/clip/_transforms.py

```diff
@@ -5,24 +5,112 @@
 
 from torchvision.transforms.v2 import functional as F
 
 logger = logging.getLogger(__name__)
 
 
+class CLIPImageTransformCore(torch.nn.Module):
+    def __init__(
+        self,
+        resample: str,
+        image_mean: Optional[List[float]],
+        image_std: Optional[List[float]],
+        tile_size: int,
+        max_image_size: int,
+        antialias: bool,
+    ):
+        super().__init__()
+        self.resample = resample
+        self.image_mean = image_mean
+        self.image_std = image_std
+        self.tile_size = tile_size
+        self.max_image_size = max_image_size
+        self.antialias = antialias
+
+    """
+    Performs torch._checks to confirm a value is within the specified lower and upper bounds.
+    Note: this is used to export the model. For eager mode usage, please disregard.
+
+    The check mitigates data dependent errors that may occur during torch.export. It installs
+    a deferred runtime assert, instead of a compile-time guard. For more context:
+    https://docs.google.com/document/d/1HSuTTVvYH1pTew89Rtpeu84Ht3nQEFTYhAX3Ypa_xJs/edit
+    """
+
+    def check_variable_bounds_for_export(self, var: int, lower: int, upper: int):
+        torch._check(var >= lower)
+        torch._check(var <= upper)
+
+    """
+    Performs the core transformations involved in CLIPImageTransform;
+    1. Resize the image to target_size.
+    2. Pad the image to canvas_size.
+    3. Normalize the image using image_mean and image_std.
+    4. Reshape the image tensor into [n, channels, tile_size, tile_size].
+    Args:
+        image (torch.Tensor): image as a 3D tensor in form [C, H, W].
+        target_size (torch.Tensor): tensor of shape [1, 2] containing the target_height and target_width for resize.
+        canvas_size (torch.Tensor): tensor of shape [1, 2] containing the canvas_height and canvas_width for padding.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]: Image tensor of shape [n, channels, tile_size, tile_size]
+            and aspect ratio tensor of shape [1, 2].
+    """
+
+    def forward(
+        self, image: torch.Tensor, target_size: torch.Tensor, canvas_size: torch.Tensor
+    ):
+        # Resize.
+        target_h, target_w = target_size.tolist()
+
+        # Checks to allow the model to export via torch.export.
+        self.check_variable_bounds_for_export(target_h, 2, self.max_image_size)
+        self.check_variable_bounds_for_export(target_w, 2, self.max_image_size)
+
+        image = resize(
+            image,
+            size=[target_h, target_w],
+            interpolation=self.resample,
+            antialias=self.antialias,
+        )
+
+        # Pad.
+        canvas_h, canvas_w = canvas_size.tolist()
+        # Checks to allow the model to export via torch.export.
+        self.check_variable_bounds_for_export(canvas_h, 2, self.max_image_size)
+        self.check_variable_bounds_for_export(canvas_w, 2, self.max_image_size)
```

Comment:
Can we just consolidate these checks together with target_h and target_w up front? Something like

```
def check_variable_bounds_for_export(self, vars: List[int]):
	for var in vars:
	        torch._check(var >= 2)
	        torch._check(var <= self.max_image_size)
```

Then we just call this as a one-liner at the beginning of the function. (Can even consider

Reply:
updated to consolidate all the checks together